lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -65,43 +71,148 @@ class UnusedJavaScript extends ByteEfficiencyAudit {
     }
 
     return {
+      unusedByIndex,
       unusedLength: unused,
       contentLength: maximumEndOffset,
     };
   }
 
   /**
-   * @param {Array<{unusedLength: number, contentLength: number}>} wasteData
-   * @param {LH.Artifacts.NetworkRequest} networkRecord
+   * @param {LH.Audit.ByteEfficiencyItem} item
+   * @param {WasteData[]} wasteData
+   * @param {import('../../computed/bundle-analysis.js').Bundle} bundle
+   * @param {ReturnType<typeof UnusedJavaScript.determineLengths>} lengths
+   */
+  static createBundleMultiData(item, wasteData, bundle, lengths) {
+    if (!bundle.script.content) return;
+
+    /** @type {Record<string, number>} */
+    const files = {};
+
+    // This is 10x slower (~320ms vs 46ms for a big map), but its correctness
+    // is much easier to reason about. The latter method gives the same counts,
+    // except it seems to have +1 byte for each file.
+    // let line = 0;
+    // let column = 0;
+    // for (let i = 0; i < bundle.script.content.length; i++) {
+    //   column += 1;
+    //   if (bundle.script.content[i] === '\n') {
+    //     line += 1;
+    //     column = 0;
+    //   }
+    //   if (wasteData.some(data => data.unusedByIndex[i] === 0)) continue;
+
+    //   // @ts-ignore: ughhhhh the tsc doesn't work for the compiled cdt lib
+    //   const mapping = bundle.map.findExactEntry(line, column);
+    //   // This can be null if the source map has gaps.
+    //   // For example, the webpack CommonsChunkPlugin emits code that is not mapped (`webpackJsonp`).
+    //   if (mapping) {
+    //     files[mapping.sourceURL] = (files[mapping.sourceURL] || 0) + 1;
+    //   }
+    // }
+
+    const lineLengths = bundle.script.content.split('\n').map(l => l.length);
+    let totalSoFar = 0;
+    const lineOffsets = lineLengths.map(len => {
+      const retVal = totalSoFar;
+      totalSoFar += len + 1;
+      return retVal;
+    });
+
+    // @ts-ignore
+    bundle.map.computeLastGeneratedColumns();
+    // @ts-ignore
+    for (const mapping of bundle.map.mappings()) {
+      let offset = lineOffsets[mapping.lineNumber];
+
+      offset += mapping.columnNumber;
+      const byteEnd = mapping.lastColumnNumber || lineLengths[mapping.lineNumber];
+      for (let i = mapping.columnNumber; i < byteEnd; i++) {
+        if (wasteData.every(data => data.unusedByIndex[offset] === 1)) {
+          files[mapping.sourceURL] = (files[mapping.sourceURL] || 0) + 1;
+        }
+        offset += 1;
+      }
+    }
+
+    const transferRatio = lengths.transfer / lengths.content;
+    const unusedFilesSizesSorted = Object.entries(files)
```

Comment:
add `top` to name since you're only taking the top 5.  in that case IMO you can drop `sorted`